--- a/Log.h
+++ b/Log.h
@@ -8,6 +8,30 @@
 #include <functional>
 #include <chrono>
 #include "Queue.h"
+#include <string>
+#include <boost/algorithm/string.hpp>
+#include <boost/lexical_cast.hpp>
+// Some platforms are missing std::to_string (issue #2743)
+#ifndef HAVE_STD_TO_STRING
+namespace std {
+template<typename V>
+inline std::string
+to_string(const V& v)
+{
+  return boost::lexical_cast<std::string>(v);
+}
+
+template<typename V>
+inline int
+stoi(const V& v)
+{
+  return boost::lexical_cast<int>(v);
+}
+
+} // namespace std
+#endif // HAVE_STD_TO_STRING
+
+
 
 enum LogLevel
 {
